bundles/org.openhab.binding.nuki/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -92,6 +92,23 @@
 				<label>Nuki ID</label>
 				<description>The decimal string that identifies the Nuki Smart Lock.</description>
 			</parameter>
+			<parameter name="deviceType" type="integer" required="true" readOnly="true">
```

Comment:
Please remove `required="true"` : as you provide a default, this parameter is in fact not "required".
And at my knowledge, the attribute readOnly is not a valid attribute. To be removed.

Reply:
Readonly works, without it user can edit the parameter in main UI. The value is required, as in the smartlock cannot function without it being specified since it must be sent with http requests. Making the deviceType `int` somewhat solves it since it can't be set to null, but still seems better to explicitly tell in the config that the propery is mandatory.

Reply:
If you create manually your device in MainUI (not through the discovery), the user must have the ability to select the device type.

While you are setting a default value for your parameter, you will always have a value, that's why "required" is not required :)
You need this default value for backward compatibility.
If you want to keep the "required" attribute, I will not block your PR but it just makes no sense IMHO.

Reply:
You actually can't create the thing manually, only using discovery (or file configuration). Marking the property as required is for documentation purposes more than anything. The idea is that optional property with default value implies that it can be explicitly set to null (which doesn't really apply here since it'll be `int`, but still).

Reply:
How MainUI can prevent creation of the thing  ? 
And why not creating the thing manually if the user prefers this way for any reason?

Reply:
It's the ` listed="false"` property on thing-type, this way if you add new thing in the UI, you'll only get Nuki bridge in the list of things you can add manually.
This is because all communication is done through the bridge and the same API endpoint that is used for discovery is also used for updating thing status (there's a /list endpoint that returns list of all connected devices and their properties). That means if the thing is not returned from this endpoint, you can't discover it but and adding it manually would not work either.

Reply:
Yes, I know this property `listed="false"`. But I am not really sure it was a good idea to use it for the reason you explained. We have in openHAB many and many bindings having the same constraint and which do not use this property. In this case, the created thing is just OFFLINE with a message explaining why. Forcing the user to use the discovery to create a thing  is not the normal/expected behaviour.
At my knowledge, this property was more to be used for thing types which become obsolete but we keep them for compatibility reasons, using this property to avoid users creating new things of such a type.

Reply:
That is interesting, I am very curious to know what are the bindings using this property.

Reply:
It is used in more bindings than I was expected: 16 bindings including the Sonos binding. So I have to change my mind on that point.
For the Sonos binding for example, the reason is that it is almost impossible for a standard user to guess the id to setup when creating the thing. I think this should be the correct trigger to use this property.

Reply:
This is similar to how for example zigbee binding works (which also use `listed` property extensively). I also did it because getting the nukiId property is not very user friendly. You can find it buried deep in some settings menu of nuki app, but it's in hexadecimal format and the bridge api uses decimal format.

Reply:
Ok.
I learned something today, this `readOnly` attribute for thing parameters.